tools/NetCorePsd1Sync/NetCorePsd1Sync/NetCoreDefinitionGenerator.cs

```diff
@@ -0,0 +1,141 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Management.Automation;
+using NetCorePsd1Sync.Model;
+using NetCorePsd1Sync.Utility;
+using static NetCorePsd1Sync.Model.PsDefinitionConstants;
+
+namespace NetCorePsd1Sync
+{
+    internal static class NetCoreDefinitionGenerator
+    {
+        private const string Psd1Extension = @".psd1";
+        private const string Psd1Filter = @"*" + Psd1Extension;
+        private const string NetCorePsd1Extension = @".Netcore" + Psd1Extension;
+        private const string NetCoreFilter = @"*" + NetCorePsd1Extension;
+
+        private static readonly Version NetCoreModuleVersion = new Version(0, 10, 0);
+        private const string NetCoreCompatiblePsEdition = "Core";
+        private static readonly Version NetCorePsVersion = new Version(5, 1);
+
+        // https://stackoverflow.com/a/25245678/294804
+        public static IEnumerable<string> GetModulePaths(string rmPath, bool ignoreExisting = false) => 
+            Directory.EnumerateDirectories(rmPath).Where(d =>
+                Directory.EnumerateFiles(d, Psd1Filter, SearchOption.TopDirectoryOnly).Any() &&
+                (ignoreExisting || !Directory.EnumerateFiles(d, NetCoreFilter, SearchOption.TopDirectoryOnly).Any()));
+
+        public static IEnumerable<string> GetDesktopFilePaths(IEnumerable<string> modulePaths) =>
```

Comment:
I pass `SearchOption.TopDirectoryOnly` to `EnumerateFiles` so that is only searches for files within that directory, and not in child directories. So, the nested `Stack` directories (like in Compute) will not be searched for psd1 files.